airflow/models/taskinstance.py

```diff
@@ -1550,11 +1550,21 @@ def _run_finished_callback(
     if callbacks:
         callbacks = callbacks if isinstance(callbacks, list) else [callbacks]
```

Comment:
Would you consider having a function to handle the logic behind the process of getting names for different instances? With it, we could simply call the function to get the name instead of checking the attributes every time when accessing the `__name__`.

```suggestion
        callbacks = callbacks if isinstance(callbacks, list) else [callbacks]
        def get_callback_name(callback):
            if hasattr(callback, "__name__"):
                return callback.__name__
            elif hasattr(callback, "__class__"):
                return callback.__class__.__name__
            else:
                return str(callback.__code__)
```